app/src/main/java/com/cornellappdev/coursegrab/CourseDetailsActivity.kt

```diff
@@ -0,0 +1,160 @@
+import androidx.recyclerview.widget.RecyclerView
+import com.cornellappdev.coursegrab.models.ApiResponse
+import com.cornellappdev.coursegrab.models.Course
+import com.cornellappdev.coursegrab.models.SearchResult
+import com.cornellappdev.coursegrab.models.Section
+import com.cornellappdev.coursegrab.networking.Endpoint
+import com.cornellappdev.coursegrab.networking.Request
+import com.cornellappdev.coursegrab.networking.addTracking
+import com.cornellappdev.coursegrab.networking.removeTracking
+import com.google.gson.reflect.TypeToken
+import kotlinx.android.synthetic.main.activity_course_details.*
+import kotlinx.android.synthetic.main.activity_search.back_btn
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class CourseDetailsActivity : AppCompatActivity() {
+    private lateinit var sectionsRecyclerView: RecyclerView
+    private lateinit var sectionsViewAdapter: RecyclerView.Adapter<*>
+    private lateinit var sectionsViewManager: RecyclerView.LayoutManager
+
+    private val preferencesHelper: PreferencesHelper by lazy {
+        PreferencesHelper(this)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_course_details)
+
+        var courseDetails: SearchResult = intent.getParcelableExtra("courseDetails") as SearchResult
+
+        header_textView.text = "${courseDetails.subject_code} ${courseDetails.course_num}"
+
+        course_title.text = courseDetails.title
+        course_details.text =
+            if (courseDetails.sections.isNotEmpty()) courseDetails.sections.first().instructors.first() else "To Be Assigned"
+
+        // Available Courses Adapter
+        sectionsViewManager = LinearLayoutManager(this@CourseDetailsActivity)
+        sectionsViewAdapter = SectionAdapter(courseDetails.sections, this@CourseDetailsActivity)
+
+        sectionsRecyclerView = findViewById<RecyclerView>(R.id.sections_recyclerview).apply {
+            layoutManager = sectionsViewManager
+            adapter = sectionsViewAdapter
+        }
+
+        back_btn.setOnClickListener { finish() }
+    }
+
+    fun addCourse(courseId: Int, context: Context) {
+        val addTracking = Endpoint.addTracking(preferencesHelper.sessionToken.toString(), courseId)
+
+        CoroutineScope(Dispatchers.Main).launch {
+            val typeToken = object : TypeToken<ApiResponse<Course>>() {}.type
+            val response = withContext(Dispatchers.IO) {
+                Request.makeRequest<ApiResponse<Course>>(
+                    addTracking.okHttpRequest(),
+                    typeToken
+                )
+            }
+
+            if (!response!!.success)
+                Toast.makeText(
+                    context,
+                    response.data.errors!![0],
+                    Toast.LENGTH_SHORT
+                ).show()
```

Comment:
We should also re-show the `track` button and hide the `remove` button if this happens

Reply:
That one's a bit difficult, since I'd have to trace back to those elements through this generic method and modify them. I'll add that as an issue for future releases.